### PR TITLE
Fix get registry url accessible in scope

### LINKED ISSUE #43: Timeout not working as expected
When running a job and the registry is not being resolved properly, the timeout is not taking the desired effect to stop the execution.

---

I'm still getting a never ending GitHub Action execution: https://github.com/oscard0m/sandbox/runs/2504376343?check_suite_focus=true. Do you want me to create a follow up issue for this?

---

_Originally posted by @oscard0m in https://github.com/gr2m/await-npm-package-version-action/issues/39#issuecomment-832299905_

### LINKED ISSUE #44: getRegistryURL not defined
@oscard0m I got an error that I think was introduced by your change, could you have a look:
https://github.com/octokit/openapi/runs/2512603282?check_suite_focus=true#step:2:41

image

I think the fix might be as simple as changing

```js
const getRegistryURL = (registry, package) => {
  const sanitizedPackageName = package.replace(/\//, "%2f");
  const registryURL = new URL(sanitizedPackageName, registry);

  return registryURL.toString();
}
```

to

```js
function getRegistryURL(registry, package) {
  const sanitizedPackageName = package.replace(/\//, "%2f");
  const registryURL = new URL(sanitizedPackageName, registry);

  return registryURL.toString();
}
```

because function declarations are hoisted, but const declarations are not

_Originally posted by @gr2m in https://github.com/gr2m/await-npm-package-version-action/issues/39#issuecomment-832979186_

### PR DESCRIPTION
**Pull request by oscard0m**
## 📝 Summary

- Make `getRegistryURL(registry, package)` function accessible from `main()` scope
- Spot the potential issue on why the timeout function was not working as expected

## ⛱ Motivation and Context

Fix #44 
Fix #43

### CODE REVIEW

**Review by gr2m — APPROVED** [review]

**Comment by gr2m** [comment]
can you check the failing tests?

File: index.js [hunk 1]
```diff
@@ -7,35 +7,32 @@ const TIMEOUT_DEFAULT = 300;
...
   return registryURL.toString();
 }
 
 async function main() {
   try {
     const package = core.getInput("package");
     const version = core.getInput("version").replace(/^v/, "");
-    const timeout = core.getInput("timeout") | TIMEOUT_DEFAULT;
-    const registry = core.getInput("registry") | "https://registry.npmjs.org"
+    const timeout = core.getInput("timeout") || TIMEOUT_DEFAULT;
```
Comment (oscard0m) [inline index.js:21]: @gr2m this fixes #43 
It was doing a logical OR instead of a fallback, so it was evaluating to `0`

File: index.js [hunk 2]
```diff
@@ -7,35 +7,32 @@ const TIMEOUT_DEFAULT = 300;
...
+    const registry = core.getInput("registry") || "https://registry.npmjs.org";
```
Comment (oscard0m) [inline index.js:22]: Same here.

**Comment by oscard0m** [comment]
> can you check the failing tests?

Hey @gr2m. I'm running tests in my local machine and they are passing. Are tests failing in your local so I can debug what's the issue?

Going to bed now but I can keep checking tomorrow. If this is creating friction, feel free to revert and I can try a new PR with all the fixes.

File: index.js (hunk 1 above)
Comment (gr2m) [reply index.js:21]: sorry I missed that!

**Review by gr2m — APPROVED** [review]

**Comment by gr2m** [comment]
yeah it passes locally for me, too. Urg, this is the worst :(

**Comment by gr2m** [comment]
I'll merge this in because the fix looks good, we can address the CI problem later

**Comment by oscard0m** [comment]
> I'll merge this in because the fix looks good, we can address the CI problem later

If it repeats again I will create an issue for it.

Thanks

**Pull request merged**